model/event/event_logger.go

```diff
@@ -26,16 +28,21 @@ func (l *DBEventLogger) LogEvent(event *EventLogEntry) error {
 	if event.Data == nil {
 		return errors.New("event log entry cannot have nil Data")
 	}
+	if len(event.ResourceType) == 0 {
+		return errors.New("event log entry has no r_type")
+	}
 	if !event.ID.Valid() {
 		event.ID = bson.NewObjectId()
 	}
-
-	found, rType := findResourceTypeIn(event.Data)
-	if !found {
-		return errors.New("event log data has no r_type")
+	if !isSubscribable(event.ResourceType) {
+		loc, _ := time.LoadLocation("UTC")
```

Comment:
should this check for error?

Reply:
if the string is "UTC" or "" it returns the UTC location, otherwise it tries to lookup the timezone in the timezone database. The lookup is what creates the errors.